Formula/typescript.rb

```diff
@@ -0,0 +1,31 @@
+require "language/node"
+
+class Typescript < Formula
+  desc "Language for application scale JavaScript development"
+  homepage "http://typescriptlang.org/"
+  url "https://registry.npmjs.org/typescript/-/typescript-2.0.3.tgz"
+  sha256 "3cfa9db91b68f95723a8f64db3dd3ee158b828d51d5a6eb6370dbca9e3045367"
+  head "https://github.com/Microsoft/TypeScript.git"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    (testpath/"test.ts").write <<-EOS.undent
+      class Test {
+        greet() {
+          return "Hello, world!";
+        }
+      };
+      var test = new Test();
+      document.body.innerHTML = test.greet();
+    EOS
+
+    system bin/"tsc", "test.ts"
+    touch testpath/"test.js", :nocreate => true
```

Comment:
It's essentially verifying that `test.ts` was properly compiled and that `tsc` outputted `test.js` as expected. The `:nocreate` ensures that if `test.js` isn't present, an exception is thrown. This can easily be removed though if it's not considered a necessary validation 😄